From patchwork Wed Oct 17 18:18:19 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wenwen Wang <wang6495@umn.edu>
X-Patchwork-Id: 10645981
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 53578181D
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Oct 2018 18:18:31 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 47D6028758
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Oct 2018 18:18:31 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 3BB5A2875C; Wed, 17 Oct 2018 18:18:31 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham
	version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id CD3B228758
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Oct 2018 18:18:30 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1728066AbeJRCPV (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 17 Oct 2018 22:15:21 -0400
Received: from mta-p7.oit.umn.edu ([134.84.196.207]:50304 "EHLO
        mta-p7.oit.umn.edu" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727082AbeJRCPV (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 17 Oct 2018 22:15:21 -0400
Received: from localhost (unknown [127.0.0.1])
        by mta-p7.oit.umn.edu (Postfix) with ESMTP id CCAFD515
        for <kvm@vger.kernel.org>; Wed, 17 Oct 2018 18:18:27 +0000 (UTC)
X-Virus-Scanned: amavisd-new at umn.edu
Received: from mta-p7.oit.umn.edu ([127.0.0.1])
        by localhost (mta-p7.oit.umn.edu [127.0.0.1]) (amavisd-new,
 port 10024)
        with ESMTP id BBuXkM3uW6ZF for <kvm@vger.kernel.org>;
        Wed, 17 Oct 2018 13:18:27 -0500 (CDT)
Received: from mail-it1-f200.google.com (mail-it1-f200.google.com
 [209.85.166.200])
        (using TLSv1.2 with cipher AES128-GCM-SHA256 (128/128 bits))
        (No client certificate requested)
        by mta-p7.oit.umn.edu (Postfix) with ESMTPS id A615A43A
        for <kvm@vger.kernel.org>; Wed, 17 Oct 2018 13:18:27 -0500 (CDT)
Received: by mail-it1-f200.google.com with SMTP id u2-v6so2983442ith.1
        for <kvm@vger.kernel.org>; Wed, 17 Oct 2018 11:18:27 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=umn.edu; s=google;
        h=from:to:cc:subject:date:message-id;
        bh=iHhie0xt3E+9d9DUTJDeOg9pQf3MJnJnRkIXolVtctI=;
        b=Zf5TtIHkzZiCqEogoyaheSCqbUaIVK06zyO94egjwwT9reagP+fbQfIr8ajlTX5xOy
         y2ll1pe5qBaXR6XS5dhk7SpA1s7d0NeIt227ZXc5xBt10S2HZrjt+Dgv5bCXrxyYfWCl
         gYuVfPPp56tYgLmgEBvcZ7RZiTAmMK8G1oFCIyqwKIR1Xp778xHSVamCLfsyt78eFI2d
         9Y1bdWQ6FgFW9qAVbp5RpK9Mtqxkfs1NfkWLh991zj/ir8HSXEXuKneQrbbWNZhpasU/
         aChGAW/t6fgxxy52LLBFHg5nBDCxP8RjfLnZNUCXIi72byd7rvHtQC1Pw5/Dui0lepa8
         15vQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=iHhie0xt3E+9d9DUTJDeOg9pQf3MJnJnRkIXolVtctI=;
        b=BqUwkDj/kpBlxHIO1FX09s9ruQ3/y2OXgx6cOEPvqOJSEfzK7/3xs1coU7VbQRTBe2
         5VSXUCZJSmC3NlQ9bbTAAlFh8H2emGmde9eO3phtj54VJglnIjjEi0gkOSTmmkmDl5KD
         cIXptQNEglmckokS4ajbyRqYlPatOF610O6TEgOvdJjabacv+pvPga9sbyaLFH2Nm23P
         0U1Nr+CPWzzPz8thdiLV3dgzPeqNdXBRBj83AsAq7LrAx5COTPBT61yPcDKRthRkqspq
         RQHR81NHo8H/71HxzKMRO1Asdha8pHsPLVO45xmXZKU447ExFA45aIGQfuX5R6vfKH3f
         v7VA==
X-Gm-Message-State: ABuFfogcYgLIfgaxLLDSeFYNUvud3bgLhGH3Rhv/u+1D2LraAb5RGW3d
        P5WAQ8IAbyXEG8PXRFN14rvgNJKzaR4ndtjub5A9Fwl7843CyQLRJ9urdl37k4YxjnC10x7cTsV
        OFSiAjvGqNdLCLD84
X-Received: by 2002:a02:13:: with SMTP id 19-v6mr20158972jaa.68.1539800307346;
        Wed, 17 Oct 2018 11:18:27 -0700 (PDT)
X-Google-Smtp-Source: 
 ACcGV61lV3oKA0N/f9lQOX6rEdZO1bET+m461j4az1UTM++HXMTrlX+RWefH7lPmU6bFfi7gYPfziw==
X-Received: by 2002:a02:13:: with SMTP id 19-v6mr20158959jaa.68.1539800307073;
        Wed, 17 Oct 2018 11:18:27 -0700 (PDT)
Received: from cs-u-cslp16.cs.umn.edu (cs-u-cslp16.cs.umn.edu.
 [134.84.121.95])
        by smtp.gmail.com with ESMTPSA id
 z186-v6sm999844itd.43.2018.10.17.11.18.25
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-SHA bits=128/128);
        Wed, 17 Oct 2018 11:18:26 -0700 (PDT)
From: Wenwen Wang <wang6495@umn.edu>
To: Wenwen Wang <wang6495@umn.edu>
Cc: Kangjie Lu <kjlu@umn.edu>,
        Alex Williamson <alex.williamson@redhat.com>,
        kvm@vger.kernel.org (open list:VFIO DRIVER),
        linux-kernel@vger.kernel.org (open list)
Subject: [PATCH] drivers/vfio: Fix an 8-byte alignment issue
Date: Wed, 17 Oct 2018 13:18:19 -0500
Message-Id: <1539800299-2787-1-git-send-email-wang6495@umn.edu>
X-Mailer: git-send-email 2.7.4
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch adds a 4-byte reserved field in the structure
vfio_eeh_pe_op to make sure that the u64 fields in the structure
vfio_eeh_pe_err are 8-byte aligned.

Signed-off-by: Wenwen Wang <wang6495@umn.edu>
---
 include/uapi/linux/vfio.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/uapi/linux/vfio.h b/include/uapi/linux/vfio.h
index 1aa7b82..3e71ded 100644
--- a/include/uapi/linux/vfio.h
+++ b/include/uapi/linux/vfio.h
@@ -729,6 +729,7 @@ struct vfio_eeh_pe_op {
 	__u32 argsz;
 	__u32 flags;
 	__u32 op;
+	__u32 __res;
 	union {
 		struct vfio_eeh_pe_err err;
 	};
